src/library/scala/collection/mutable/ListBuffer.scala

```diff
@@ -114,6 +114,25 @@ class ListBuffer[A]
     this
   }
 
+  // Overridden for performance
+  override final def addAll(xs: IterableOnce[A]): this.type = {
```

Comment:
there are a couple of special cases - adding a List and adding a ListBuffer
With a small tweak to copyElems we could structurally share the content and make the addition O(1) for time and space

copyElems would need to change to know where the sharing started for an added List. If `aliased` was a ref that would work though I think

Reply:
I don't believe that the typical use cases would benefit from those special cases. `ListBuffer.addAll` is commonly called from `List.from`, which already reuses the input seq is also a `List`.